src/ebonite/build/runner/catalog.py

```diff
@@ -0,0 +1,30 @@
+from typing import Dict, Type, Union
+
+from ebonite.build.runner.base import RunnerBase
+from ebonite.build.runner.docker import DockerRunner
+from ebonite.core.objects import RuntimeInstance
+
+
+class RunnersCatalog:
+    """
+    Catalog for all known :class:`.RunnerBase` implementations
+    """
+
+    _alias_runners: Dict[str, RunnerBase] = {
```

Comment:
now the only option to add new runner is to change ebonite sources. I propose to add a way to dynamically define new Runner implementations - something like Analyzers. and also we can move all docker-related code to 'ebonite.ext.docker' for that matter

Reply:
This looks like a reasonable idea. But it could take a fair amount of time and therefore should be implemented if we plan to have other runner implementations soon. I'm not Product Owner but for now I don't think this is true.

Reply:
this is important not because we are planning to add runner implementations, but because we must provide a way for others to add their custom runners. i agree this could be another PR, but no need to postpone that until the end of times)